cns/restserver/restserver.go

```diff
@@ -1723,3 +1740,218 @@ func (service *HTTPRestService) deleteHostNCApipaEndpoint(w http.ResponseWriter,
 	err = service.Listener.Encode(w, &response)
 	log.Response(service.Name, response, response.Response.ReturnCode, ReturnCodeToString(response.Response.ReturnCode), err)
 }
+
+// Check if the network is joined
+func (service *HTTPRestService) isNetworkJoined(networkID string) bool {
+	namedLock.LockAcquire(stateJoinedNetworks)
+	defer namedLock.LockRelease(stateJoinedNetworks)
+
+	if service.state.joinedNetworks == nil {
+		service.state.joinedNetworks = make(map[string]struct{})
+	}
+
+	_, exists := service.state.joinedNetworks[networkID]
+
+	return exists
+}
+
+// Set the network as joined
+func (service *HTTPRestService) setNetworkStateJoined(networkID string) {
+	namedLock.LockAcquire(stateJoinedNetworks)
+	defer namedLock.LockRelease(stateJoinedNetworks)
+
+	service.state.joinedNetworks[networkID] = struct{}{}
+}
+
+// Join Network by calling nmagent
+func (service *HTTPRestService) joinNetwork(
+	networkID string,
+	joinNetworkURL string) (*http.Response, error, error) {
+	var err error
```

Comment:
I will expose to check if the network is joined or not, for debugging purposes.

Reply:
I have added log indicating if the network is joined successfully and if that nw is marked 'joined' in memory